cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java

```diff
@@ -1486,8 +1547,25 @@ String getPrompt() {
         }
 
         if (isBatchMode()) {
-            buffer.append(" #");
+            if (getConfig().isColorOutput()) {
+                String batchPrompt = " #] ";
+                batchPrompt = Util.formatWorkflowPrompt(batchPrompt);
+                return buffer.append(batchPrompt).toString();
+            } else {
+                buffer.append(" #");
+            }
         }
+
+        if (isWorkflowMode()) {
+            if (getConfig().isColorOutput()) {
+                String workflowPrompt = " ...] ";
```

Comment:
We should keep the prompt as it used to be.

Reply:
Same for batch mode?

Reply:
yes, we could later log an RFE for this prompt enhancement.

Reply:
I've just done a presentation to my team about this feature and I got positive feedback on the prompt enhancement. @jfdenise  is there some particular reason to do it in different RFE?

Reply:
I wouldn't want that we would break some tests that would parse the prompt. If QE is OK to include this enhancement in this RFE then I agree.

Reply:
It is OK from tests POV. I already ran wildfly-core tests and there were no problems with it. I will run also other related tests. If something fails because of this change it is no problem to fix the test.

I've discussed this with colleagues and we found use case where it could cause problems to users.

```
./jboss-cli.sh < file-with-commands
```

If you execute commands by given way, the prompt is printed to the output together with commands. If users parse such output, this could break their scripts.

@bmaxwell Do you know whether this could break users automation? This RFE changes CLI prompt when `if`, `try` or `for` commands are used. It adds `...` into the prompt. So it looks in following way.

```
[standalone@embedded  /] try
[standalone@embedded  / ...] command-1
[standalone@embedded  / ...] command-2
[standalone@embedded  / ...] finally
[standalone@embedded  / ...] command-3
[standalone@embedded  / ...] end-try
```

Reply:
I also thought at it, but got the feeling that this would be not common enough to freeze the prompt, I might be wrong.

Reply:
I don't see a problem with it as well. Should we go ahead?